### ISSUE TITLE
android: adblock reddit promoted ads (again)

### ISSUE THREAD

**Issue by sbernecchia** [brave/brave-browser#29013]
## Description 
I just updated brave on my phone (android) to latest version 1.49.122, which should have support for adblock cosmetic filter has()
I was hoping to never see again the well known "promoted" ads but I still see one of them

## Steps to reproduce 

   1. difficult to reproduce, because these ads appears random while you browse reddit
   2. I have seen one of these promoted ads only inside a post, not in the main list
   3. Example:  https://www.reddit.com/r/vivaldibrowser/comments/11ppl8c/vivaldi_57292160_mac_delete_all_cookies_does_not/
   4. 

## Actual result 
![promoted ads again](https://user-images.githubusercontent.com/73037836/224663372-b735b758-f529-49ba-a4c5-6c7444210bad.jpg)

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version? 
- Can you reproduce this issue with the current Play Store Beta version? 
- Can you reproduce this issue with the current Play Store Nightly version?

## Device details
   - Install type (ARM, x86): arm
   - Device type (Phone, Tablet, Phablet): phone
   - Android version: 12
   
## Brave version
v1.49.122

### Website problems only
- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards?
- Is the issue reproducible on the latest version of Chrome?

**Comment by Emi-TheDhamphirInLoveUnderTheFrozenStar** [comment]
Yes, Desktop and Android adblocker have the same features, and just like that, that means you need to set the Adblocker to Aggressive in Settings, because if not, default lists have the 1p vs 3p logic Brave built on it (custom lists and rules don't).

Standard mostly filters 3p, Aggressive filters 1p and 3p like any normal adblocker, and reddit promoted ads are 1p, so that's the reason.

Promoted reddit posts don't even need :has(), they are a normal class cosmetics rule in Easylist: `.promotedlink`, so any adblocker supports that. 9gag promoted posts were the ones that needed a `:has()`, well, `:-abp-has()`, but not reddit.

**Comment by sbernecchia** [comment]
no, it's already set to aggressive, and still I'm viewing those ads.

**Comment by sbernecchia** [comment]
another one:

![image](https://user-images.githubusercontent.com/73037836/226165176-5e5a0fe2-756f-49ab-9976-7e2ec8507679.png)

**Comment by ryanbr** [comment]
Does adding `reddit.com###adLinkBar` into brave://adblock ?  I couldn't reproduce this

**Comment by sbernecchia** [comment]
thank you for the rule, but the ad is still not blocked.

reproducing this issue is difficult, the ad appears randomly.
after I added your rule I had to keep browsing and opening various posts until I saw the ad again.

how can I help?
saving a page could help? (you could see the html code)
how can I save a page in brave for android?
can I do using brave://inspect page?
